README.md

```diff
@@ -203,7 +203,35 @@ COPY bin/my-wag-service /usr/bin/my-wag-service
 
   * If you don't have a context to pass to a Wag function you have two options
     * context.Background() - use this when this is the creator of the request chain, like a test or a top-level service.
-    * context.TODO() - use this when you haven't been passed a context from a caller yet, but you expect the caller to send you one at some point.  
+    * context.TODO() - use this when you haven't been passed a context from a caller yet, but you expect the caller to send you one at some point.
+
+
+### DynamoDB Codegen
+
+  * Wag can auto-generate server code to save models to DynamoDB if you specify the `x-db` extension on a schema:
+  ```yaml
+definitions:
+  Thing:
+    x-db:
+      AllowOverwrites: false
+      DynamoDB:
+        KeySchema:
+          - AttributeName: name
+            KeyType: HASH
+          - AttributeName: version
+            KeyType: RANGE
+    type: object
+    properties:
+      name:
+        type: string
+      version:
+        type: integer
+  ```
+  The above will generate a `db` package with code to load/persist `Thing` objects from/to DynamoDB.
+  * `AllowOverwrites` specifies whether the auto-generated `Save` method should succeed if the object already exists.
+  * `DynamoDB` specifies the configuration for a DyanmoDB table for the schema.
+     It follows the format of the [`AWS::DynamoDB::Table`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html) CloudFormation resource.
+     Currently it supports a subset of the configuration allowed there.
 
```

Comment:
Might be nice to link a real working example from here (workflow-manager)

Reply:
Because it would show invoking the model from the controller, rather than just the auto-generated code that doesn't invoke the DB methods yet.

Reply:
Another thing I'm curious to see in action is `CreateTables`... is it just useful for the unit tests? do we think expect our apps to generate tables at runtime or continue to have them created out of band? generating tables at runtime would be especially awesome in dev, but makes less sense in prod. Also configuration like autoscaling isn't captured here.

Reply:
👍 will add link to wfm once this is merged and wfm is updated

Right now `CreateTables` is only used in test, and the only configuration supported is primary `KeySchema`. In the future I could see
- Supporting more config options that influence the codegen in the swagger spec: TTL, secondary indexes (these are things that wfm requires)
- Figuring out some strategy for creating tables per-environment. I lean towards some mechanism where catapult includes the dynamo tables in the cloudformation template for an app. How that gets from the swagger spec to the template, I'm not sure...

Reply:
Also in the future I could see catapult/ark owning setting autoscaling config. Here I lean towards something not configured in code, but rather changed on the fly via `ark`, e.g. `ark scale`